models/multimeter.cpp

```diff
@@ -91,27 +98,56 @@ nest::Multimeter::Parameters_::set( const DictionaryDatum& d,
   const Buffers_& b )
 {
   if ( b.has_targets_
-    && ( d->known( names::interval ) || d->known( names::record_from ) ) )
+    && ( d->known( names::interval ) || d->known( names::offset )
+         || d->known( names::record_from ) ) )
+  {
     throw BadProperty(
-      "The recording interval and the list of properties to record "
-      "cannot be changed after the multimeter has been connected to "
-      "nodes." );
+      "The recording interval, the interval offset and the list of properties "
+      "to record "
+      "cannot be changed after the multimeter has been connected to nodes." );
+  }
 
   double v;
   if ( updateValue< double >( d, names::interval, v ) )
   {
     if ( Time( Time::ms( v ) ) < Time::get_resolution() )
+    {
       throw BadProperty(
-        "The sampling interval must be at least as long "
-        "as the simulation resolution." );
+        "The sampling interval must be at least as long as the simulation "
+        "resolution." );
+    }
 
     // see if we can represent interval as multiple of step
     interval_ = Time::step( Time( Time::ms( v ) ).get_steps() );
     if ( std::abs( 1 - interval_.get_ms() / v ) > 10
         * std::numeric_limits< double >::epsilon() )
+    {
+      throw BadProperty(
+        "The sampling interval must be a multiple of the simulation "
+        "resolution" );
+    }
+  }
+
+  if ( updateValue< double >( d, names::offset, v ) )
+  {
+    // if offset is different from the default value (0), it must be at least
+    // as large as the resolution
+    if ( v != 0 && Time( Time::ms( v ) ) < Time::get_resolution() )
+    {
+      throw BadProperty(
+        "The offset for the sampling interval must be at least as long as the "
+        "simulation resolution." );
+    }
+
+    // see if we can represent offset as multiple of step
+    offset_ = Time::step( Time( Time::ms( v ) ).get_steps() );
+    if ( std::abs( 1 - offset_.get_ms() / v ) > 10
+        * std::numeric_limits< double >::epsilon() )
+    {
```

Comment:
applied both changes along with the fix for @mschmidt87 's example and edited the unit test to account for this. Let me know if anything else needs to be done.